src/backend/distributed/deparser/qualify_statistics_stmt.c

```diff
@@ -68,8 +72,13 @@ QualifyDropStatisticsStmt(Node *node)
 
 		if (stat->schemaname == NULL)
 		{
-			Oid schemaOid = RangeVarGetCreationNamespace(stat);
-			stat->schemaname = get_namespace_name(schemaOid);
+			Oid statsOid = get_statistics_object_oid(objectNameList, true);
```

Comment:
would it make sense to use `dropStatisticsStmt->missing_ok` instead of the hardcoded value?

Reply:
I'm OK with both. Because if the object is not found, and missing_ok is set to false, the error message will be the same in both approaches.

Reply:
@agedemenli there are many more places in the code where this should be changed accordingly.